aws-distro-opentelemetry-node-autoinstrumentation/src/patches/aws/services/lambda.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Attributes, Span, SpanKind, Tracer } from '@opentelemetry/api';
+import {
+  AwsSdkInstrumentationConfig,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWS_ATTRIBUTE_KEYS } from '../../../aws-attribute-keys';
+import { RequestMetadata, ServiceExtension } from '../../../third-party/otel/aws/services/ServiceExtension';
+
+export class LambdaServiceExtension implements ServiceExtension {
+  requestPreSpanHook(request: NormalizedRequest, _config: AwsSdkInstrumentationConfig): RequestMetadata {
+    const requestFunctionName = request.commandInput?.FunctionName;
+    const resourceMappingId = request.commandInput?.UUID;
+
+    const spanKind: SpanKind = SpanKind.CLIENT;
+    let spanName: string | undefined;
+
+    const spanAttributes: Attributes = {};
+
+    let functionName = requestFunctionName;
+
+    if (requestFunctionName) {
+      if (requestFunctionName.startsWith('arn:aws:lambda:')) {
+        const split = requestFunctionName.split(':');
+        functionName = split[split.length - 1];
+      }
+
+      spanAttributes[AWS_ATTRIBUTE_KEYS.AWS_LAMBDA_FUNCTION_NAME] = functionName;
```

Comment:
Previously, there was some bug with Lambda:Function:Name so we decided to temporarily remove support for it.

- https://github.com/aws-observability/aws-otel-python-instrumentation/pull/280
- https://github.com/aws-observability/aws-otel-python-instrumentation/pull/281

I'm not sure if that issue has been resolved yet. Should we remove it here as well?

cc: @mxiamxia